### DISCUSSION TITLE
Forms with Xstate, Vue and Joi

### DISCUSSION THREAD

**Discussion by thomas-kuhlmann** [statelyai/xstate#2034] [Q&A]
Hi,

I'm currently using Vue 2 and Vuelidate for validating my forms. I've started moving my application's business logic to xstate a few weeks ago (love it, it has been an absolute lifesaver!). Initially, I kept form management relatively light touch - I only managed overall form state with xstate, but not individual fields ( I kept those in Vue and validated them with Vuelidate).

I am now thinking about moving the entire form (including fields) to xstate and use Joi for validation. I started thinking this through and it seems to be a pretty daunting exercise - has anybody already done this? I'd rather like to avoid reinventing the wheel if possible (especially when it comes to i.e. input field states), so if somebody has some good, working examples that can handle more complex scenarios and include managing the states, I'd love to get some inspiration!

**Comment by davidkpiano** [comment]
Here's a relevant discussion: https://github.com/davidkpiano/xstate/discussions/1851
